src/mysql-util/src/schemas.rs

```diff
@@ -7,57 +7,241 @@
+                            .transpose()
+                            .map_err(|_| MySqlError::UnsupportedDataType {
+                                column_type,
+                                qualified_table_name: format!("{:?}.{:?}", schema_name, table_name),
+                                column_name: column_name.clone(),
+                            })?,
+                    }
+                }
+                "char" => ScalarType::Char {
+                    length: character_maximum_length
+                        .and_then(|f| Some(CharLength::try_from(f)))
+                        .transpose()
+                        .map_err(|_| MySqlError::UnsupportedDataType {
+                            column_type,
+                            qualified_table_name: format!("{:?}.{:?}", schema_name, table_name),
+                            column_name: column_name.clone(),
+                        })?,
+                },
+                "varchar" => ScalarType::VarChar {
+                    max_length: character_maximum_length
+                        .and_then(|f| Some(VarCharMaxLength::try_from(f)))
+                        .transpose()
+                        .map_err(|_| MySqlError::UnsupportedDataType {
+                            column_type,
+                            qualified_table_name: format!("{:?}.{:?}", schema_name, table_name),
+                            column_name: column_name.clone(),
+                        })?,
+                },
+                "text" | "mediumtext" | "longtext" => ScalarType::String,
+                "binary" | "varbinary" | "tinyblob" | "blob" | "mediumblob" | "longblob" => {
+                    ScalarType::Bytes
+                }
+                // TODO: Implement other types
+                _ => Err(MySqlError::UnsupportedDataType {
+                    column_type,
+                    qualified_table_name: format!("{:?}.{:?}", schema_name, table_name),
+                    column_name: column_name.clone(),
+                })?,
             };
-            columns.push((
-                ordinal_position,
-                MySqlColumnDesc {
-                    name,
-                    data_type,
-                    nullable: is_nullable == "YES",
+            columns.push(MySqlColumnDesc {
+                name: column_name,
+                column_type: ColumnType {
+                    scalar_type,
+                    nullable: &is_nullable == "YES",
                 },
-            ));
+            })
+        }
+
+        // Query for primary key and unique constraints that do not contain expressions / functional key parts.
+        let index_rows = conn.exec::<(String, String), _, _>("
+            WITH indexes AS (
+                SELECT
+                    index_name,
+                    group_concat(column_name ORDER BY seq_in_index separator ', ') AS columns,
+                    group_concat(expression ORDER BY seq_in_index separator ', ') AS expressions
```

Comment:
Fair point -- how do you feel about using `JSON_ARRAYAGG` instead? https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_json-arrayagg

It seems easier to do the grouping and filtering (of indexes with expressions) in the database instead, and `JSON_ARRAYAGG` seems to avoid the potential escaping / string concatenation issues that we might otherwise hit.

Reply:
discussed offline -- Petros suggested a better approach:

```
SELECT
    index_name,
    column_name
FROM information_schema.statistics outer
WHERE
    table_schema NOT IN ('information_schema', 'mysql', 'performance_schema', 'sys')
    AND NOT EXISTS (
        SELECT 1
        FROM information_schema.statistics inner
        WHERE outer.index_name = inner.index_name AND inner.column_name IS NULL
    )
    AND non_unique = 0
    AND table_name = ?
    AND table_schema = ?
ORDER BY index_name, seq_in_index
```